block_element.go

```diff
@@ -475,3 +476,34 @@ func NewRadioButtonsBlockElement(actionID string, options ...*OptionBlockObject)
 		Options:  options,
 	}
 }
+
+// NumberInputBlockElement creates a field where a user can enter number
+// data.
+// Number input elements are currently only available in modals.
+//
+// More Information: https://api.slack.com/reference/block-kit/block-elements#number
+type NumberInputBlockElement struct {
```

Comment:
[note] focus_on_load field is missing, but I think it is OK to submit another PR for this one after merging this PR.